src/inmanta/data/__init__.py

```diff
@@ -5498,14 +5500,20 @@ async def get_versions(cls, environment: uuid.UUID, start: int = 0, limit: int =
         return versions
 
     async def delete_cascade(self) -> None:
-        async with self.get_connection() as con:
-            async with con.transaction():
-                # Delete all code associated with this version
-                await Code.delete_all(connection=con, environment=self.environment, version=self.version)
-
-                # Delete ConfigurationModel and cascade delete on connected tables
-                await self.delete(connection=con)
+        await Compile.delete_all(environment=self.environment, version=self.version)  # Triggers cascading delete report table
+        await Code.delete_all(environment=self.environment, version=self.version)
```

Comment:
I wonder why don't we have a foreign key reference for `Compile` and `Code` to `ConfigurationModel.version` the same way we have it for `UnknownParameter`  or `DryRun` ? i.e.
```sql
ADD CONSTRAINT unknownparameter_environment_version_fkey FOREIGN KEY (environment, version) REFERENCES public.configurationmodel(environment, version) ON DELETE CASCADE;
```

Reply:
I was wondering the same. That's why I created the following ticket yesterday: https://github.com/inmanta/inmanta-core/issues/6581

Reply:
I didn't implement it right away, because adding this migration script might be tricky. The database might be in an inconsistent state, because the foreign key constraint was not enforced. I didn't want to pollute this PR with that code and I didn't want to wait merging this PR, because this PR is a blocker for the release. Fixing the foreign key constraint isn't.